rest/route/host_agent.go

```diff
@@ -1254,6 +1262,10 @@ func (h *hostAgentEndTask) Run(ctx context.Context) gimlet.Responder {
 			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
 		})
 	}
+	ctx = utility.ContextWithAttributes(ctx, []attribute.KeyValue{
```

Comment:
That makes sense to me but would there be a better way than to set it twice like I'm doing?

Reply:
I can't think of a way to avoid having to do both.
The way we propagate the attributes down is with [a span processor that the OTel SDK calls when a span is started](https://github.com/evergreen-ci/utility/blob/4d91d64f59fb653893f0f0b0f301971ce050b389/otel.go#L24). If we add attributes to the context after the span is started they aren't added to the span.